File: LeetCode/LeetCode-CN-CPP/2020六月每日一题/9. 回文数.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) {
            return false;
        }
        int origin_num = x;
        long reverse_num = 0;
        while(x != 0) {
            reverse_num *= 10;
            reverse_num += x%10;
            x /= 10;
        }
        if(reverse_num == origin_num) {
            return true;
        }
        return false;
    }
};

File: LeetCode/LeetCode-CN-JAVA/9. 回文数.java
class Solution {
    public boolean isPalindrome(int x) {
        // 小于0肯定不是回文数
        // 最后一位为0，那么第一位必须为0，所以此时x应该为0
        if(x < 0 || (x%10 == 0 && x!=0)) {
            return false;
        }

        int reversedNumber = 0;
        while(x > reversedNumber) {
            reversedNumber = reversedNumber * 10 + x%10;
            x /= 10;
        }
        return reversedNumber == x || reversedNumber/10 == x;
    }
}